### PR TITLE
improve methodolgic of adding new car and beatify code

### PR DESCRIPTION
**Pull request by idobi111**

### CODE REVIEW

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 1]
```diff
@@ -117,19 +113,104 @@ private Vehicle getCarFromUserByType(eVehicleType i_Type)
...
+        {
+            Console.Write("Please insert cargo size: ");
+            float cargoSize = float.Parse(Console.ReadLine());
+            return cargoSize;
+        }
+
+        private static float getFuelLeftFromUser()
+        {
+            Console.Write("Please insert the fuel left: ");
+            float powerLeft = float.Parse(Console.ReadLine());
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: No exception if user inputs a wrong value

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 2]
```diff
@@ -117,19 +113,104 @@ private Vehicle getCarFromUserByType(eVehicleType i_Type)
...
+            return model;
+        }
+
         private eMotorcycleLicense getBikeLicense()
         {
-            Console.WriteLine("Please insert the bike license type\n1. A1\n2. A2\n3. AA\n4. B1");
-            Enum.TryParse(Console.ReadLine(), out eMotorcycleLicense o_bikeLicense);
+            Display.BikeLicense();
+
+            if (Enum.TryParse(Console.ReadLine(), out eMotorcycleLicense o_bikeLicense))
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: Change o_bikeLicense to motorcycleLicense

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 3]
```diff
@@ -117,19 +113,104 @@ private Vehicle getCarFromUserByType(eVehicleType i_Type)
...
 
         private bool getTransportingHazardousMaterial()
         {
             bool hazardousMaterial;
-            Console.WriteLine("Is transporting hazardous material? ");
-            Console.WriteLine("1. Yes");
-            Console.WriteLine("2 No ");
+            Display.TransportingHazardousMaterial();
             string transportingHazardousMaterialStr = Console.ReadLine();
             if (transportingHazardousMaterialStr == "1")
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: What about enum?

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 4]
```diff
@@ -210,15 +301,22 @@ private List<Wheel> createNewWheel(int i_NumOfWheels)
...
-            Console.Write("Please insert the max PSI: ");
-            float userMaxPSI = float.Parse(Console.ReadLine());
             Console.Write("Please insert the manufaturer: ");
             string userManufaturer = Console.ReadLine();
             Console.Write("Please insert the current PSI: ");
             float userCurrentPSI = float.Parse(Console.ReadLine());
-            return new Wheel(userMaxPSI, userManufaturer, userCurrentPSI);
+            if (!isValidCurrentPSI(userCurrentPSI, i_MaxPsi))
+            {
+                throw new ArgumentException("The enterence current PSI bigger then the Max allowed PSI");
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: "Current PSI cannot be more than MaxPSI"

Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: Dםמק

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 2 above)
Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: Done

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 1 above)
Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: Done

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 3 above)
Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: Done

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 1 above)
Comment (Ronenii) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: This should check if user tries to input more than the MAX Liters amount, and also if the user tries to input some random string

Comment (Ronenii) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: In general the user doesn't input the powerLeft, he inputs the number of liters of fuel he wants in the car. the powerLeft is a calculation done during the creation of the vehicle

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 5]
```diff
@@ -117,25 +113,124 @@ private Vehicle getCarFromUserByType(eVehicleType i_Type)
...
         private bool getTransportingHazardousMaterial()
         {
             bool hazardousMaterial;
-            Console.WriteLine("Is transporting hazardous material? ");
-            Console.WriteLine("1. Yes");
-            Console.WriteLine("2 No ");
-            string transportingHazardousMaterialStr = Console.ReadLine();
-            if (transportingHazardousMaterialStr == "1")
+            Display.TransportingHazardousMaterial();
+            if (Enum.TryParse(Console.ReadLine(), out eTransportingHazardousMaterial o_TransportingHazardousMaterial))
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: remove o_, this is only done in input parameters in functions

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 6]
```diff
@@ -148,8 +243,20 @@ private bool getTransportingHazardousMaterial()
 
         private eCarColor getColorFromUser()
         {
-            Console.WriteLine("Please insert the bike license type\n1. White\n2. Black\n3. Yellow\n4. Red");
-            Enum.TryParse(Console.ReadLine(), out eCarColor o_carColor);
+            Display.ColorMenu();
+
+            if (Enum.TryParse(Console.ReadLine(), out eCarColor o_carColor))
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: remove o_, this is only done in input parameters in functions

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs [hunk 7]
```diff
@@ -210,15 +315,22 @@ private List<Wheel> createNewWheel(int i_NumOfWheels)
...
         }
 
-        private Wheel createWheelFromUser()
+        private Wheel createWheelFromUser(float i_MaxPsi)
         {
-            Console.Write("Please insert the max PSI: ");
-            float userMaxPSI = float.Parse(Console.ReadLine());
             Console.Write("Please insert the manufaturer: ");
             string userManufaturer = Console.ReadLine();
             Console.Write("Please insert the current PSI: ");
```
Comment (Ronenii) [inline B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs:340]: need to check also that user doesn't add negative PSI

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 6 above)
Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: ok 
but there are a lot of this in the code

Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: i will do it in other bracnh
fix all

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 5 above)
Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs]: i will do it in other branch delete all

File: B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs (hunk 7 above)
Comment (idobi111) [reply B23 Ex03 Ronen 319047718 Ido 315942193/VehicleFactory.cs:340]: ADD

**Pull request merged**